AGS_oM/FuzzyObjectResult.cs

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BH.oM.Base;
+
+namespace BH.oM.Adapters.AGS
+{
+    [Description("A result class containing objects, scores and indexes from the fuzzy matching.")]
+    public class FuzzyObjectResult : BHoMObject, IFuzzyResult, IImmutable
+    {
+        /***************************************************/
+        /****            Public Properties              ****/
+        /***************************************************/
+
+        [Description("A list of objects resulting from the fuzzy matching algorithm.")]
+        public virtual List<BHoMObject> Results { get; }
+
+        [Description("A list of scores resulting from the fuzzy matching algorithm.")]
+        public virtual List<int> Scores { get; }
+
+        [Description("A list of indexes resulting from the fuzzy matching algorithm.")]
+        public virtual List<int> Indexes { get; }
```

Comment:
In order to keep result, score, and index together, it would be better to have a FuzzyResult object only containing one of each. Otherwise, it doesn't solve the issue we discussed previously. So, something like this:

```c#
public class FuzzyResult<T> : BHoMObject, IResult
{
    public virtual T Result { get; }

    public virtual int Score { get; }

    public virtual int Index { get; }
}
```

Notice I also used the `IResult` interface to be more consistent with the rest of the BHoM. 
I also made the class generic as it removed the need to have multiple classes and an interface covering them (i.e. `FuzzyStringResult` and `IFuzzyResult`).